### PR TITLE
Use HDF5 files for quantity logs.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Switch from GSD to HDF5 for thermodynamic logs.

## Motivation and context

Reduce the time needed to run the analysis workflow steps.

## How has this been tested?

- [x] Test entire workflow run.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/ContributorAgreement.md).
- [x] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
I ran the complete workflow on Great Lakes. I didn't measure before/after timings, but the analysis step feels faster. The total output file size is also much smaller: 81M for 4 replicates.

File: hoomd_validation/hard_disk.py [hunk 1]
```diff
@@ -165,20 +165,20 @@ def make_mc_simulation(job,
     sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(mc, quantities=['translate_moves'])
-    logger_gsd.add(sdf, quantities=['betaP'])
-    logger_gsd.add(compute_density, quantities=['density'])
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/hard_disk.py:167]: The comment above this line is no longer true.

File: hoomd_validation/hard_sphere.py [hunk 2]
```diff
@@ -160,20 +160,20 @@ def make_mc_simulation(job,
     sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(mc, quantities=['translate_moves'])
-    logger_gsd.add(sdf, quantities=['betaP'])
-    logger_gsd.add(compute_density, quantities=['density'])
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/hard_sphere.py:162]: see above comment.

File: hoomd_validation/hard_sphere.py [hunk 3]
```diff
@@ -311,21 +311,20 @@ def run_nec_sim(job, device, complete_filename):
...
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(mc,
-                   quantities=[
-                       'translate_moves', 'particles_per_chain',
-                       'virial_pressure'
-                   ])
-    logger_gsd.add(compute_density, quantities=['density'])
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/hard_sphere.py:312]: see above comment.

File: hoomd_validation/simple_polygon.py [hunk 4]
```diff
@@ -178,12 +178,12 @@ def make_mc_simulation(job,
     sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(mc, quantities=['translate_moves'])
-    logger_gsd.add(sdf, quantities=['betaP'])
-    logger_gsd.add(compute_density, quantities=['density'])
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/simple_polygon.py:180]: see above comment.

File: hoomd_validation/patchy_particle_pressure.py [hunk 5]
```diff
@@ -291,13 +291,13 @@ def make_mc_simulation(job,
     sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(mc, quantities=['translate_moves'])
-    logger_gsd.add(sdf, quantities=['betaP'])
-    logger_gsd.add(compute_density, quantities=['density'])
-    logger_gsd.add(patches, quantities=['energy'])
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/patchy_particle_pressure.py:293]: see above comment.

File: hoomd_validation/lj_union.py [hunk 6]
```diff
@@ -568,20 +568,20 @@ def make_mc_simulation(job,
     compute_density = ComputeDensity()
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(lj_jit_potential, quantities=['energy'])
-    logger_gsd.add(mc, quantities=['translate_moves', 'rotate_moves'])
-    logger_gsd.add(compute_density)
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/lj_union.py:570]: see above comment.

File: hoomd_validation/lj_fluid.py [hunk 7]
```diff
@@ -525,20 +525,20 @@ def make_mc_simulation(job,
     compute_density = ComputeDensity()
 
     # log to gsd
-    logger_gsd = hoomd.logging.Logger(categories=['scalar', 'sequence'])
-    logger_gsd.add(lj_jit_potential, quantities=['energy'])
-    logger_gsd.add(mc, quantities=['translate_moves'])
-    logger_gsd.add(compute_density)
+    logger = hoomd.logging.Logger(categories=['scalar', 'sequence'])
```
Comment (tommy-waltmann) [inline hoomd_validation/lj_fluid.py:527]: see above comment.

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
The changes look mostly good. There is just one comment which appears to have been copy/pasted in each validation test file which is no longer true.

**Review by tommy-waltmann — APPROVED** [review]

**Pull request merged**